cmd/build/v2/checker/checker.go

```diff
@@ -0,0 +1,92 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package checker
+
+import (
+	"context"
+
+	"github.com/okteto/okteto/cmd/build/v2/environment"
+	"github.com/okteto/okteto/cmd/build/v2/metadata"
+	"github.com/okteto/okteto/pkg/build"
+	"github.com/okteto/okteto/pkg/env"
+	"github.com/okteto/okteto/pkg/log/io"
+	"github.com/okteto/okteto/pkg/registry"
+)
+
+const (
+	parallelCheckStrategyEnvVar = "OKTETO_BUILD_CHECK_STRATEGY_PARALLEL"
+)
+
+// ImageTagger defines the interface for managing image tags and references.
+type ImageTagger interface {
+	GetGlobalTagFromDevIfNeccesary(tags, namespace, registryURL, buildHash string, ic registry.ImageCtrl) string
+	GetImageReferencesForTag(manifestName, svcToBuildName, tag string) []string
+	GetImageReferencesForDeploy(manifestName, svcToBuildName string) []string
+}
+
+// SmartBuildController defines the interface for smart build operations.
+type SmartBuildController interface {
+	GetBuildHash(buildInfo *build.Info, service string) string
+	CloneGlobalImageToDev(globalImage, svcImage string) (string, error)
+}
+
+// CheckStrategy defines the interface for different cache checking strategies.
+type CheckStrategy interface {
+	CheckServicesCache(ctx context.Context, manifestName string, buildManifest build.ManifestBuild, svcsToBuild []string) (cachedServices []string, notCachedServices []string, err error)
+	CloneGlobalImagesToDev(manifestName string, buildManifest build.ManifestBuild, images []string) error
```

Comment:
Why don't we encapsulate this under the same operation to outside of this? I mean, this could return only the list of non-cached images, and make sure the rest of them are copied, optimizing the process, instead of having 2 different "parallel" implementations (one for the check, and another one for the clone)